lib/logstash/outputs/webhdfs.rb

```diff
@@ -0,0 +1,282 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/outputs/base"
+require "stud/buffer"
+
+# Summary: Plugin to send logstash events to to files in HDFS via webhdfs 
+# restapi.  
+#
+# This plugin only has a mandatory dependency on the webhdfs gem from 
+# Kazuki Ohta and TAGOMORI Satoshi (@see: https://github.com/kzk/webhdfs).
+# Optional dependencies are zlib and snappy gem. 
+# No jars from hadoop are needed, thus reducing configuration and compatibility
+# problems.
+#
+# If you get an error like:
+#
+#     Max write retries reached. Exception: initialize: name or service not known {:level=>:error}
+#
+# make sure, that the hostname of your namenode is resolvable on the host running logstash. When creating/appending
+# to a file, webhdfs somtime sends a 307 TEMPORARY_REDIRECT with the HOSTNAME of the machine its running on.
+#
+# USAGE:
+# This is an example of logstash config:
+#
+# [source,ruby]
+# ----------------------------------
+# webhdfs {
```

Comment:
Expanded the example in https://github.com/dstore-dbap/logstash-output-webhdfs-1/commit/1710bf284ead4c613a7afe757c1604a2aeabb693